Tab change bug, DataView show/hide, bold titles

CelineMP opened the pull request:
Fixed this bug: https://github.com/amosproj/amos2024ss04-building-information-enhancer/issues/212

There were two problems:
- Clicking on "…" bubbled up to parent and always led to tab change
- Tab Option Menu used the opened tab's id. So I made a new tab ID that is set when "…" is clicked

Corgam commented:
I am so happy that you managed to fix it! I just have one comment that the Info popup option is not working correctly

CelineMP commented:
Oh no true, the info is broken! 
Yes the border looks bad. It shouldn't even be focusable, it should just change to the tab.

CelineMP commented:
@Corgam I fixed the three mentioned things now:
- Info PopUp bug
- You can click everywhere on the Tab to change, not only the text
- Removed focus outline

CelineMP commented:
Now this PR also includes:
- https://github.com/amosproj/amos2024ss04-building-information-enhancer/issues/247
- https://github.com/amosproj/amos2024ss04-building-information-enhancer/issues/248

The functionality is now there. But where exactly the hide/show button should be can be discussed next sprint. For now I put it next to the "new tab" button.

Lucas-Nan commented:
emils proposed variant doesnt support hiding the whole thing so nicely

Corgam commented:
> @Lucas-Nan
> 
> I implemented your feedback. But I personally prefer a different background color.
> 
> <img alt="Bildschirmfoto 2024-06-11 um 17 46 31" width="428" src="https://private-user-images.githubusercontent.com/58625805/338655730-5a92a2bc-aa64-48c1-af85-6de9f27ed899.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgxMjExNjMsIm5iZiI6MTcxODEyMDg2MywicGF0aCI6Ii81ODYyNTgwNS8zMzg2NTU3MzAtNWE5MmEyYmMtYWE2NC00OGMxLWFmODUtNmRlOWYyN2VkODk5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjExVDE1NDc0M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIzNGFlYTViZTRiZjJhMmM5OTk2ZmZhYmJhZTBjY2FjZmUxNDFiNWI4ODJlY2RhMjY3MmMxNTJjODhjYTBhNDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oDHJKkATs4erYffRcqVSZpvQUCe3c5_U6VHFL6DHdsU">
> @Corgam
> 
> The problem is: when you put the icon into the DataView, where is it when the DataView is hidden?

I like it how it is right now after the changes, maybe just use the double arrows and I would be really happy haha

CelineMP commented:
> I like it how it is right now after the changes, maybe just use the double arrows and I would be really happy haha

Done!
<img width="295" alt="Bildschirmfoto 2024-06-11 um 17 52 09" src="https://github.com/amosproj/amos2024ss04-building-information-enhancer/assets/58625805/dafc9e95-ed8e-4f67-a6a3-24e3cd6f807c">
